Mesh_3/include/CGAL/lloyd_optimize_mesh_3.h

```diff
@@ -21,50 +21,147 @@
+ * when the function \f$ x^2\f$ is interpolated on the mesh domain
+ * using a piecewise linear function which is linear
+ * in each cell of the Voronoi diagram of the mesh vertices.
+ *
+ * The optimizer `lloyd_optimize_mesh_3()` works in iterative steps.
+ * At each iteration, mesh vertices are moved into
+ * positions that bring to zero the energy gradient
+ * and the Delaunay triangulation is updated.
+ * Vertices on the mesh boundaries are handled
+ * in a special way so as to preserve an accurate
+ * representation of the domain boundaries.
+ *
+ * \tparam C3T3 a model of the concept `MeshComplex_3InTriangulation_3`.
+ * \tparam MD a model of the concept `MeshDomain_3`.
+ * \tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+ *
+ * @param c3t3 the initial mesh that will be modified by the algorithm to represent the final optimized mesh.
+ * @param domain the domain used to create the `c3t3` parameter
+ * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamNBegin{time_limit}
+ *     \cgalParamDescription{to set up, in seconds, a CPU time limit after which the optimization process is stopped.
+ *                           This time is measured using `CGAL::Real_timer`. 0 means that there is no time limit.}
+ *     \cgalParamType{`double`}
+ *     \cgalParamPrecondition{`time_limit >= 0`}
+ *     \cgalParamDefault{0}
+ *   \cgalParamNEnd
+ *   \cgalParamNBegin{max_iteration_number}
+ *     \cgalParamDescription{limit on the number of performed iterations. 0 means that there is
+ *                           no limit on the number of performed iterations.}
+ *     \cgalParamPrecondition{`max_iteration_number >=0`}
+ *     \cgalParamType{`int`}
+ *     \cgalParamDefault{0}
+ *   \cgalParamNEnd
+ *   \cgalParamNBegin{freeze_bound}
+ *     \cgalParamDescription{designed to reduce running time of each optimization iteration.
+ *                           Any vertex that has a displacement less than a given fraction of the length
+ *                           of its shortest incident edge, is frozen (i.e.\ is not relocated).
+ *                           The parameter `freeze_bound` gives the threshold ratio.
+ *                           If it is set to 0, freezing of vertices is disabled.}
+ *     \cgalParamPrecondition{`0<= freeze_bound <=1`}
+ *     \cgalParamType{`double`}
+ *     \cgalParamDefault{0.001}
+ *   \cgalParamNEnd
+ *   \cgalParamNBegin{convergence}
+ *     \cgalParamDescription{threshold ratio of stopping criterion based on convergence: the optimization process is stopped
+ *                           when at the last iteration the displacement of any vertex is less than
+ *                           a given fraction of the length of the shortest edge incident to that vertex.}
+ *     \cgalParamPrecondition{`0 <=convergence <= 1`}
+ *     \cgalParamType{`double`}
+ *     \cgalParamDefault{0.02}
```

Comment:
The current code is 0.001, dunno which one is wrong

Reply:
Fixed